lib/dat-manager.js

```diff
@@ -0,0 +1,138 @@
+      this.emit('update')
+    })
+  }
+  watch (dat) {
+    dat.on('update', () => this.emit('update'))
+    return dat
+  }
+  get () {
+    return this.dats
+  }
+  create (dir, cb) {
+    const dat = Dat({ dir })
+    dat.owner = true
+    dat.dir = dir
+    dat.share(err => {
+      if (err) return cb(err)
+      this.persist(dat, err => {
+        if (err) return cb(err)
+        this.dats.push(this.watch(this.activate(dat)))
+        this.emit('update')
+        cb()
+      })
+    })
+  }
+  download (key, cb) {
+    const dir = `${root}/${key}`
+    fs.mkdir(dir, err => {
+      if (err) return cb(err)
+      const dat = Dat({ dir, key })
+      dat.owner = false
+      dat.dir = dir
+      this.persist(dat, err => {
+        if (err) return cb(err)
+        dat.download()
+        this.dats.push(this.watch(this.activate(dat)))
+        this.emit('update')
+        cb()
+      })
+    })
+  }
+  load ({ key, owner, dir }) {
+    if (Buffer.isBuffer(key)) key = encoding.encode(key)
+    const dat = Dat({ key, dir })
+    dat.dir = dir
+    dat.owner = owner
+    if (dat.owner) {
+      dat.share(err => {
+        if (err) dat.emit('error', err)
+      })
+    } else {
+      dat.download(err => {
+        if (err) dat.emit('error', err)
+      })
+    }
+    return dat
+  }
+  loadAll (cb) {
+    const dats = []
+    db.createReadStream({
+      gt: ['archive', null],
+      lt: ['archive', undefined]
+    })
+    .on('data', data => {
+      dats.push(this.activate(this.load(data.value)))
+    })
+    .on('error', err => cb(err))
+    .on('end', () => cb(null, dats))
+  }
+  persist ({ key, owner, dir }, cb) {
+    db.put(['archive', key], {
+      key: encoding.encode(key),
+      owner,
+      dir
+    }, cb)
+  }
+  activate (dat) {
+    dat.open(err => {
+      if (err) throw err
+      dat.listStream = dat.archive.list({ live: true })
+      dat.listStream.on('data', entry => {
+        if (entry.name !== 'dat.json') return
+        collect(dat.archive.createFileReadStream('dat.json'), (err, raw) => {
+          if (err) return console.error('failed to read dat.json', err)
+          let json
+          try {
+            json = JSON.parse(raw.toString())
+          } catch (err) {
+            console.error('failed to parse dat.json', err)
+            return
+          }
+          dat.title = json.title
+          dat.emit('update')
+        })
+      })
+      dat.hyperStats = Stats({
+        archive: dat.archive,
+        db: sub(db, `${encoding.encode(dat.key)}-stats`)
+      })
+      dat.hyperStats.on('update', () => dat.emit('update'))
+      dat.on('files-counted', () => dat.emit('update'))
+      dat.on('archive-finalized', () => dat.emit('update'))
+      dat.on('archive-updated', () => dat.emit('update'))
+      dat.on('download-finished', () => dat.emit('update'))
+      dat.on('swarm-update', () => dat.emit('update'))
+    })
+    return dat
+  }
+  remove (dat, cb) {
```

Comment:
Remove might be different from pause, right? I might want to stop the swarm and updates, which would mean "deactivating" - what do you think about having a separate function for that and then remove calls deactivate then removes it from the list